tools/validator.py

```diff
@@ -100,12 +100,22 @@ def CheckFile(self, filename):
           self._artifact_registry.RegisterDefinition(artifact_definition)
         except KeyError:
           logging.warning(
-              u'Duplicate artifact definition: {0:s} in file: {1:s}'.format(
+              'Duplicate artifact definition: {0:s} in file: {1:s}'.format(
                   artifact_definition.name, filename))
           result = False
 
         for source in artifact_definition.sources:
-          if source.type_indicator == (
+          if source.type_indicator in (
+              definitions.TYPE_INDICATOR_FILE, definitions.TYPE_INDICATOR_PATH):
+            if source.supported_os == [definitions.SUPPORTED_OS_WINDOWS]:
+              if source.separator != '\\':
```

Comment:
This check kind of implies that all artifacts that support Windows need to have the separator set to "\". If this is the case, why have the separator user defined at all?

Reply:
> why have the separator user defined at all?

There are various different types of path separators: https://en.wikipedia.org/wiki/Path_(computing) and artifacts are intended to self contain the necessary information in the definition. We could create a WINDOWS_PATH but IMHO that does not improve this. 

One additional thing the validator could check if the path contains more `\` than '/' for Windows. This change is a short term fix to ensure existing definition (for at least Windows) are explicit about their path segment separator.

Reply:
Right I get the intention behind this. What I'm saying is that it will be confusing that when an artifact actually defines a path separator that is not '\', let's say that artifact wants to use '/' instead. In that case, the validation will complain with "Missing path separator...", which doesn't make sense.

Reply:
I'll add the additional check for the number of `\` versus '/' for now, but I somewhat agree with you on one side but on another side also would like this validated.